packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts

```diff
@@ -125,45 +120,33 @@ function getFrontEndConfig(authParameters: any) {
     });
   }
 
-  if (loginMechanism.length) {
-    frontEndConfig.loginMechanism = loginMechanism;
-  }
+  const signupAttributes: string[] = authParameters?.requiredAttributes
+    ? authParameters.requiredAttributes.map((att: string) => att.toUpperCase())
+    : [];
```

Comment:
if the default should be an empty array, I'd recommend:
```suggestion
const signupAttributes = (authParameters?.requiredAttributes || []).map((att: string) => att.toUpperCase());
```